src/main/java/com/lishid/openinv/ConfigUpdater.java

```diff
@@ -0,0 +1,161 @@
+package com.lishid.openinv;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.bukkit.Material;
+import org.bukkit.configuration.ConfigurationSection;
+
+import com.lishid.openinv.utils.UUIDUtil;
+
+public class ConfigUpdater {
+    private final OpenInv plugin;
+
+    private static final int CONFIG_VERSION = 2;
+
+    public ConfigUpdater(OpenInv plugin) {
+        this.plugin = plugin;
+    }
+
+    private int getConfigVersion() {
+        return plugin.getConfig().getInt("config-version", 1);
+    }
+
+    private boolean isConfigOutdated() {
+        return getConfigVersion() < CONFIG_VERSION;
+    }
+
+    public void checkForUpdates() {
+        if (isConfigOutdated()) {
+            plugin.getLogger().info("[Config] Update found! Performing update...");
+            updateConfig();
+        } else {
+            plugin.getLogger().info("[Config] Update not found. Config is already up-to-date.");
+        }
+    }
+
+    private void updateConfig() {
+        // Get the old config settings
+        int itemOpenInvItemId = plugin.getConfig().getInt("ItemOpenInvItemID", 280);
+        boolean checkForUpdates = plugin.getConfig().getBoolean("CheckForUpdates", true);
+        boolean notifySilentChest = plugin.getConfig().getBoolean("NotifySilentChest", true);
+        boolean notifyAnyChest = plugin.getConfig().getBoolean("NotifyAnyChest", true);
+
+        Map<UUID, Boolean> anyChestToggles = null;
+        Map<UUID, Boolean> itemOpenInvToggles = null;
+        Map<UUID, Boolean> silentChestToggles = null;
+
+        if (plugin.getConfig().isSet("AnyChest")) {
+            anyChestToggles = updateAnyChestToggles();
+        }
+
+        if (plugin.getConfig().isSet("ItemOpenInv")) {
+            itemOpenInvToggles = updateItemOpenInvToggles();
+        }
+
+        if (plugin.getConfig().isSet("SilentChest")) {
+            silentChestToggles = updateSilentChestToggles();
+        }
+
+        // Clear the old config
+        for (String key : plugin.getConfig().getKeys(false)) {
+            plugin.getConfig().set(key, null);
+        }
+
+        // Set the new config options
+        plugin.getConfig().set("config-version", CONFIG_VERSION);
```

Comment:
Could probably set Config config = plugin.getConfig() and use config later on. Minor detail though.